src/main/java/net/neoforged/neoforge/transfer/resources/FluidResource.java

```diff
@@ -0,0 +1,336 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.transfer.resources;
```

Comment:
I wonder if this class would not be a better fit for the `net.neoforged.neoforge.fluid(s)` package. After all it's a fairly general concept.